tool/plugin/lib/edit.dart

```diff
@@ -4,14 +4,271 @@
+        'FlutterProjectModel model(@NotNull NewModuleModel project,',
+      );
+      source = source.replaceAll(
+        'mySharedModel.getValue().project().setValue(project);',
+        'mySharedModel.getValue().project().setValue(project.getProject().getValue());',
+      );
+      processedFile.writeAsStringSync(source);
+      return original;
+    } else {
+      return null;
+    }
+  }
+}
+
+/// Apply all the editCommands applicable to a given BuildSpec.
+Future<int> applyEdits(BuildSpec spec, Function compileFn) async {
+  // Handle skipped files.
+  for (String file in spec.filesToSkip) {
+    var entity =
+        FileSystemEntity.isFileSync(file) ? File(file) : Directory(file);
+    if (entity.existsSync()) {
+      await entity.rename('$file~');
+      if (entity is File) {
+        var stubFile = File('${file}_stub');
+        if (stubFile.existsSync()) {
+          await stubFile.copy('$file');
+        }
+      }
+    }
+  }
+
+  var edited = <EditCommand, String>{};
+  try {
+    editCommands.forEach((edit) {
+      var source = edit.convert(spec);
+      if (source != null) {
+        edited[edit] = source;
+        appliedEditCommands.add(edit);
+      }
+    });
+
+    return await compileFn.call();
+  } finally {
+    // Restore sources.
+    edited.forEach((edit, source) {
+      log('Restoring ${edit.path}');
+      edit.restore(source);
+    });
+
+    // Restore skipped files.
+    for (var file in spec.filesToSkip) {
+      var name = '$file~';
+      var entity =
+          FileSystemEntity.isFileSync(name) ? File(name) : Directory(name);
+      if (entity.existsSync()) {
+        await entity.rename(file);
+      }
+    }
+  }
+}
+
+/// Make some changes to a source file prior to compiling for a specific IDE.
+abstract class EditCommand {
+  String convert(BuildSpec spec);
+
+  void restore(String source) {
+    var processedFile = File(path);
+    processedFile.writeAsStringSync(source);
+  }
+
+  String get path;
+}
+
+/// Single substitution in a file for one or more IDE versions.
+class Subst extends EditCommand {
+  @override
+  String path;
+  String initial;
+  String replacement;
+  List<String> versions;
+
+  Subst({this.versions, this.initial, this.replacement, this.path, version})
+      : assert(initial != null),
+        assert(replacement != null),
+        assert(path != null) {
+    if (version != null) {
+      // Disallow both version and versions keywords.
+      assert(versions == null);
+      versions = [version];
+    }
+  }
+
+  @override
+  String convert(BuildSpec spec) {
+    if (versionMatches(spec)) {
+      var processedFile = File(path);
+      var source = processedFile.readAsStringSync();
+      var original = source;
+      source = source.replaceAll(initial, replacement);
+      processedFile.writeAsStringSync(source);
+      return original;
+    } else {
+      return null;
+    }
+  }
+
+  String toString() {
+    return "Subst(path: $path, versions: $versions)";
```

Comment:
this could use the expression shorthand: `==>`

Reply:
Done!